dev-util/buildbot-worker/buildbot-worker-3.11.0.ebuild

```diff
@@ -0,0 +1,139 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYPI_NO_NORMALIZE=1
+PYTHON_COMPAT=( python3_{10..12} )
+inherit readme.gentoo-r1 systemd distutils-r1 pypi
+
+DESCRIPTION="BuildBot Worker (slave) Daemon"
+HOMEPAGE="https://buildbot.net/
+	https://github.com/buildbot/buildbot
+	https://pypi.org/project/buildbot-worker/"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~sparc ~amd64-linux ~x86-linux"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+SRC_URI="${SRC_URI} https://dev.gentoo.org/~zorry/patches/buildbot/buildbot-worker-${PV}-remove_py27.tar.gz"
+
+RDEPEND="
+	acct-user/buildbot
+	!<dev-util/buildbot-3.0.0
+	>=dev-python/autobahn-0.16.0[${PYTHON_USEDEP}]
+	>=dev-python/msgpack-0.6.0[${PYTHON_USEDEP}]
+	>=dev-python/twisted-18.7.0[${PYTHON_USEDEP}]
+"
+BDEPEND="
+	test? (
+		${RDEPEND}
+		dev-python/parameterized[${PYTHON_USEDEP}]
+		dev-python/psutil[${PYTHON_USEDEP}]
+	)
+"
+
+DOC_CONTENTS="The \"buildbot\" user and the \"buildbot_worker\" init script has been added
+to support starting buildbot_worker through Gentoo's init system. To use this,
+execute \"emerge --config =${CATEGORY}/${PF}\" to create a new instance.
+Set up your build worker following the documentation, make sure the
+resulting directories are owned by the \"buildbot\" user and point
+\"${ROOT}/etc/conf.d/buildbot_worker.myinstance\" at the right location.
+The scripts can	run as a different user if desired."
+
+src_prepare() {
+	# Remove shipped windows start script
+	sed -e "/'buildbot_worker_windows_service=buildbot_worker.scripts.windows_service:HandleCommandLine',/d" \
+		-i setup.py || die
+	# applay remove py 2.7 patch
+	eapply "${WORKDIR}/buildbot-worker-${PV}-remove_py27.patch"
```

Comment:
This is... pretty obscure, and not obvious that it actually consists of a backport from upstream git master. 1100 lines worth of patch, in fact, which is a bit on the difficult side to guarantee it will continually rebase.

... which disappointingly, despite promises that it would be in the next release, is not in fact in the next release after all. I've pinged upstream: https://github.com/buildbot/buildbot/pull/6997#issuecomment-1913958150

Reply:
It was suppose to be in next release.  Thanks for pinging upstream. Can add more info in the ebuild and patch. Buildbot is the last package that depend on future in gentoo.

Reply:
Yeah I've been working pretty hard to get future into a position to be treecleaned so I know all about it. :D

buildbot rejected my conservative patch to move python2 support from future to six, because they said they'd release this patch for fully dropping python2 in the next minor release... now they did not do so. Baffling and disappointing.